integrations/redis_test.go

```diff
@@ -0,0 +1,76 @@
+package integrations_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-redis/redis/v7"
+
+	"github.com/reddit/baseplate.go/integrations"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+func TestRedisSpanHook(t *testing.T) {
+	ctx := context.Background()
+	span := tracing.StartSpanFromThriftContext(ctx, "foo")
+	ctx, _ = span.SetServerSpan(ctx)
+	hooks := integrations.RedisSpanHook{ClientName: "redis"}
+	cmd := redis.NewStatusCmd("ping")
+
+	t.Run(
+		"Before/AfterProcess",
+		func(t *testing.T) {
+			ctx, err := hooks.BeforeProcess(ctx, cmd)
+			if err != nil {
+				t.Fatalf("Unexpected error: %s", err)
+			}
+			childSpan := tracing.GetChildSpan(ctx)
+			if childSpan == nil {
+				t.Fatalf("'childSpan' is 'nil'")
+			}
+			if childSpan.Name != "redis.ping" {
+				t.Fatalf("Incorrect span name '%s'", childSpan.Name)
+			}
+
+			if err = hooks.AfterProcess(ctx, cmd); err != nil {
+				t.Fatalf("Unexpected error: %s", err)
+			}
+			childSpan = tracing.GetChildSpan(ctx)
+			if childSpan == nil {
+				t.Fatalf("'childSpan' is 'nil'")
+			}
+			if err = childSpan.End(ctx, nil); err != tracing.ErrSpanAlreadyEnded {
```

Comment:
nit: Since go1.13, prefer to use `errors.Is(err, tracing.ErrSpanAlreadyEnded)` instead of direct comparison (`Is` also does unwrapping).